subgraphs/erc20/schema.graphql

```diff
@@ -0,0 +1,293 @@
+# Subgraph Schema: ERC20 Tokens
+# Version: 1.0.0
+
+#############################
+####### Token Metadata ######
+#############################
+
+type Token @entity {
+  " Smart contract address of the token "
+  id: ID!
+
+  " Name of the token, mirrored from the smart contract "
+  name: String!
+
+  " Symbol of the token, mirrored from the smart contract "
+  symbol: String!
+
+  " The number of decimal places this token uses, default to 18 "
+  decimals: Int!
+
+  " If token transfers are paused, derived from paused event "
+  isPaused: Boolean
+
+  " Token Holders Balance "
+  holdersBalance: [AccountBalance!]! @derivedFrom(field: "token")
+
+  " Number of token holders "
+  holderCount: BigInt!
+
+  " Total number of token transfer transactions "
+  transferCount: BigInt!
+
+  " Total number of token mint transactions "
+  mintCount: BigInt!
+
+  " Total number of token burn transactions "
+  burnCount: BigInt!
+
+  " Total token supply "
+  totalSupply: BigDecimal!
+
+  " Total token burned "
+  totalBurned: BigDecimal!
+
+  " Total token minted "
+  totalMinted: BigDecimal!
+
+  " List of token transfers "
+  transfers: [Transfer!]! @derivedFrom(field: "token")
+
+  ##### Snapshots #####
+
+  " Daily usage metrics for this Token "
+  dailyUsageMetrics: [UsageMetricsDailySnapshot!]! @derivedFrom(field: "token")
+
+  " Hourly usage metrics for this Token "
+  hourlyUsageMetrics: [UsageMetricsHourlySnapshot!]!
+    @derivedFrom(field: "token")
+
+  dailyTokenSnapshot: [TokenDailySnapshot!]! @derivedFrom(field: "token")
+}
+
+###############################
+####### Token Timeseries ######
+###############################
+
+type UsageMetricsDailySnapshot @entity {
+  " { Token Address } - {# of days since Unix epoch time} "
+  id: ID!
+
+  " Token this snapshot is associated with "
+  token: Token!
+
+  " # of accounts holding this token "
+  dailyHolderCount: Int!
+
+  " # of unique daily active users "
+  dailyActiveUsers: Int!

```

Comment:
should we rename users to holders since we are using holders elsewhere?

Reply:
yeah that make sense